src/App.tsx

```diff
@@ -37,19 +39,23 @@ class App extends React.Component<Props, State> {
     return this.blog.getBlogPosts()
       .then((posts) => {
         this.setState({ posts: posts.reverse() });
+
       });
   }
 
   addBlogPost(title: string, content: string) {
     return this.blog.addBlogPost(title, content)
```

Comment:
this.setState({ isWaitingOnPost: true }) needs to happen before the addBlogPost line.